app/static/src/app/components/GraphSettings.vue

```diff
@@ -0,0 +1,75 @@
+<template>
+    <div>
+        <div class="log-scale-y-axis form-check form-check-inline">
+            <label class="form-check-label">Log scale y axis</label>
+            <input type="checkbox" class="form-check-input" style="vertical-align: bottom" v-model="logScaleYAxis" />
+        </div>
+        <div class="lock-y-axis form-check form-check-inline">
+            <label class="form-check-label">Lock y axis</label>
+            <input type="checkbox" class="form-check-input" style="vertical-align: bottom" v-model="lockYAxis" />
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed } from "vue";
+import { useStore } from "vuex";
+import { GraphsMutation } from "../store/graphs/mutations";
+
+export default defineComponent({
+    name: "GraphSettings",
+    props: {
+        graphIndex: {
+            type: Number,
+            required: false,
+            default: -1
+        },
+        fitPlot: {
+            type: Boolean,
+            required: true
+        }
+    },
+    setup(props) {
+        const store = useStore();
+        const settings = computed(() =>
+            props.fitPlot ? store.state.graphs.fitGraphSettings : store.state.graphs.config[props.graphIndex].settings
+        );
+        const logScaleYAxis = computed({
+            get() {
+                return settings.value.logScaleYAxis;
+            },
+            set(newValue) {
+                if (props.fitPlot) {
+                    store.commit(`graphs/${GraphsMutation.SetFitLogScaleYAxis}`, newValue);
+                } else {
+                    store.commit(`graphs/${GraphsMutation.SetLogScaleYAxis}`, {
+                        graphIndex: props.graphIndex,
+                        value: newValue
+                    });
+                }
+            }
+        });
+
+        const lockYAxis = computed({
+            get() {
+                return settings.value.lockYAxis;
+            },
+            set(newValue) {
+                if (props.fitPlot) {
+                    store.commit(`graphs/${GraphsMutation.SetFitLockYAxis}`, newValue);
+                } else {
+                    store.commit(`graphs/${GraphsMutation.SetLockYAxis}`, {
+                        graphIndex: props.graphIndex,
+                        value: newValue
+                    });
+                }
+            }
+        });
+
+        return {
+            logScaleYAxis,
+            lockYAxis
+        };
+    }
+});
+</script>
```

Comment:
It occurs to me that the GraphSettings component checks props.fitPlot in three places to do slightly different `computed` behavior. So an alternative refactor would be to define the fit plot as a component that inherits (extends) from the GraphSettings, and overwrites these three computeds. That idea doesn't provide much at the moment, probably, but I thought I'd mention it.